.github/workflows/smoke-test.yml

```diff
@@ -0,0 +1,23 @@
+name: Smoke Test
+
+on:
+  workflow_dispatch:
+
+concurrency:
+  group: smoke-test
+  cancel-in-progress: true
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  test:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: actions/checkout@v3.5.3
+      - uses: Swatinem/rust-cache@v2.5.1
+      - run: cargo build --manifest-path crates_io_smoke_test/Cargo.toml
+      - run: cargo run --manifest-path crates_io_smoke_test/Cargo.toml --quiet
+        env:
+          CARGO_REGISTRY_TOKEN: ${{ secrets.STAGING_SMOKE_TEST_TOKEN }}
```

Comment:
@rust-lang/infra to make this work we need to add a repository secret with a staging API token, scoped to `publish-update` and the `crates-staging-test-tb` crate.

should I generate such a token myself? do you want to do that? should I invite any users to my test crate?